src/ucp/api/ucp.h

```diff
@@ -170,6 +171,44 @@ enum ucp_dt_type {
 
 
 /**
+ * @ingroup UCP_WORKER
+ * @brief UCP worker parameters field mask.
+ *
+ * The enumeration allows specifying which fields in @ref ucp_worker_params_t are
+ * present. It is used for the enablement of backward compatibility support.
+ */
+enum ucp_worker_params_field {
+    UCP_PARAM_FIELD_THREAD_MODE     = UCS_BIT(0), /* thread_mode */
+    UCP_PARAM_FIELD_CPU_SET         = UCS_BIT(1), /* cpu_set */
+};
+
+
+/**
+ * @ingroup UCP_WORKER
+ * @brief UCP parameters for ucp worker creation
+ *
+ */
+struct ucp_worker_params {
+    /**
+      * cpu_set affinity to use 
+      */
+    ucs_cpu_set_t                 cpu_set;
+
+    /**
+     * Mask of valid fields in this structure, using bits from @ref ucp_worker_params_field.
+     * Fields not specified in this mask would be ignored.
+     * Provides ABI compatibility with respect to adding new fields.
+     */
+    uint64_t                      field_mask;
+
+     /**
+      *  thread mode for ucp worker
+      */
+    ucs_thread_mode_t             thread_mode;
+};
```

Comment:
Do we really want to define `ucp_worker_params_t` in other `ucp/api/ucp_def.h` file by `typedef` instead simply define it in this place?